src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -142,6 +148,11 @@ public final class StringConcatFactory {
     // code for runtime checks. Take care any changes or additions here are
     // reflected there as appropriate.
 
+    @SuppressWarnings("removal")
+    private static boolean isGenerateInlineCopy() {
+        return GENERATE_INLINE_COPY && System.getSecurityManager() == null;
```

Comment:
This security manager hack is unfortunate. Likely some bootstrapping issue due use of the classfile API, which might be pulling in lambdas in the paths travelled by this patch.